src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -431,6 +438,124 @@ private OpenApiOperation CreateBaseOperation(string description, List<OpenApiTag
             return operation;
         }
 
+        /// <summary>
+        /// This method creates a list of OpenApiParameter objects for the input parameters of a stored procedure.
+        /// A input parameter will be marked REQUIRED if default value is not available.
+        /// </summary>
+        private static List<OpenApiParameter> CreateStoredProcedureInputParameters(StoredProcedureDefinition spDefinition)
+        {
+            List<OpenApiParameter> parameters = new();
+
+            foreach ((string paramKey, ParameterDefinition parameterDefinition) in spDefinition.Parameters)
+            {
+                parameters.Add(
+                    GetOpenApiQueryParameter(
+                        name: paramKey,
+                        description: "Input parameter for stored procedure parameter",
+                        required: !parameterDefinition.HasConfigDefault,
```

Comment:
As pointed out by Rohann, it's difficult to get the default values of arguments from metadata, so if default values are provided in the config, `HasConfigDefault` would be made true, hence it wouldn't require values to be explicitly provided when making a request

Reply:
With this, we are still unable to catch the case if the config doesnt specify the value, whereas the stored proc definition has it set as optional.. we would end up saying this parameter is required. 

But thats not the case, I think the `required` argument should always be set to false, we cant really say if its really required or not. It may not be correctly indicative.

Reply:
make sense, will update it.